autogluon_zeroshot/contexts/context.py

```diff
@@ -0,0 +1,329 @@
+        df_results, df_results_by_dataset, df_raw, df_metadata = load_results(
+            results=self.result,
+            results_by_dataset=self.results_by_dataset,
+            raw=self.raw,
+            metadata=self.task_metadata,
+            require_tid_in_metadata=self.task_metadata is not None,
+        )
+        return df_results, df_results_by_dataset, df_raw, df_metadata
+
+    def load_comparison(self) -> pd.DataFrame:
+        return load_pd.load(self.comparison)
+
+    def load_zpp(self, zsc, lazy_format=False):
+        self._assert_exists(self.zs_pp, name='zs_pp')
+        self._assert_exists(self.zs_gt, name='zs_gt')
+        zeroshot_pred_proba, zeroshot_gt, zsc = load_zeroshot_input(
+            path_pred_proba=self.zs_pp,
+            path_gt=self.zs_gt,
+            zsc=zsc,
+            lazy_format=lazy_format,
+        )
+        return zeroshot_pred_proba, zeroshot_gt, zsc
+
+    def load_configs(self):
+        return load_configs(self.configs)
+
+
+class BenchmarkContext:
+    def __init__(self,
+                 *,
+                 folds: List[int],
+                 benchmark_paths: BenchmarkPaths,
+                 name: str = None,
+                 description: str = None,
+                 date: str = None,
+                 s3_download_map: Dict[str, str] = None,
+                 ):
+        self.folds = folds
+        self.benchmark_paths = benchmark_paths
+        self.name = name
+        self.description = description
+        self.date = date
+        self.s3_download_map = s3_download_map
+
+    @classmethod
+    def from_paths(cls,
+                   *,
+                   folds: List[int],
+                   name: str = None,
+                   description: str = None,
+                   date: str = None,
+                   s3_download_map: Dict[str, str] = None,
+                   **paths):
+        return cls(folds=folds,
+                   name=name,
+                   description=description,
+                   date=date,
+                   s3_download_map=s3_download_map,
+                   benchmark_paths=BenchmarkPaths(**paths))
+
+    def download(self, include_zs: bool = True, exists: str = 'raise', dry_run: bool = False):
+        print(f'Downloading files for {self.name} context... '
+              f'(include_zs={include_zs}, exists="{exists}", dry_run={dry_run})')
+        if dry_run:
+            print(f'\tNOTE: `dry_run=True`! Files will not be downloaded.')
+        assert self.s3_download_map is not None, \
+            f'self.s3_download_map is None: download functionality is disabled'
+        file_paths_expected = self.benchmark_paths.get_file_paths(include_zs=include_zs)
+
+        file_paths_to_download = [f for f in file_paths_expected if f in self.s3_download_map]
+        if len(file_paths_to_download) == 0:
+            print(f'WARNING: Matching file paths to download is 0! '
```

Comment:
Not technically an error if the only files that need to be download are zs_pp and zs_gt, and `include_zs=False`. This could result in an empty file_paths_to_download.

I expect this won't be a big issue, and we can revise later if needed. This is more just highlighting a potentially bugged s3_download_map.